packages/auth/src/providers/cognito/utils/signInStore.ts

```diff
@@ -49,22 +64,74 @@ const signInReducer: Reducer<SignInState, SignInAction> = (state, action) => {
 				username: action.value,
 			};
 		case 'SET_INITIAL_STATE':
-			return defaultState();
+			return initializeState();
+		case 'RESET_STATE':
+			return getDefaultState();
 		default:
 			return state;
 	}
 };
 
-function defaultState(): SignInState {
+const isExpired = (expiryDate: string | null): boolean => {
+	if (!expiryDate) {
+		return true;
+	}
+	const expiryTimestamp = Number(expiryDate);
+	const currentTimestamp = Date.now();
+
+	return expiryTimestamp <= currentTimestamp;
+};
+
+const clearPersistedSignInState = () => {
+	for (const key in signInStateKeys) {
+		syncSessionStorage.removeItem(key);
+	}
+};
+
+// Clear saved sign in states from both memory and Synced Session Storage
+export function cleanActiveSignInState(): void {
+	signInStore.dispatch({ type: 'RESET_STATE' });
+	clearPersistedSignInState();
+}
+
+const getDefaultState = (): SignInState => ({
+	username: undefined,
+	challengeName: undefined,
+	signInSession: undefined,
+});
+
+// Hydrate signInStore from Synced Session Storage
+const initializeState = (): SignInState => {
+	if (!syncSessionStorage) {
+		return getDefaultState();
+	}
+
+	const expiry = syncSessionStorage.getItem(signInStateKeys.expiry);
+	if (isExpired(expiry)) {
+		logger.warn('Sign-in session expired');
+		clearPersistedSignInState();
+
+		return getDefaultState();
+	}
+
+	const username =
+		syncSessionStorage.getItem(signInStateKeys.username) ?? undefined;
+
+	const challengeName = (syncSessionStorage.getItem(
+		signInStateKeys.challengeName,
+	) ?? undefined) as ChallengeName;
+	const signInSession =
+		syncSessionStorage.getItem(signInStateKeys.signInSession) ?? undefined;
+
 	return {
-		username: undefined,
-		challengeName: undefined,
-		signInSession: undefined,
+		username,
+		challengeName,
+		signInSession,
 	};
-}
+};
 
 const createStore: Store<SignInState, SignInAction> = reducer => {
-	let currentState = reducer(defaultState(), { type: 'SET_INITIAL_STATE' });
+	let currentState = reducer(getDefaultState(), { type: 'SET_INITIAL_STATE' });
```

Comment:
Thinking twice, we can probably dispatch a different action if the session was already persisted. That way we can addressed the concerns above

```suggestion
const action = isStatePersisted() ? {type:'SET_PERSISTED_STATE'} : { type: 'SET_INITIAL_STATE' };
let currentState = reducer(getDefaultState(), action );
```

Reply:
It's not a super necessary refactor imo, but I am onboard with the change. Essentially, taking the logic out of the mutation function and making a determination of which type of action to dispatch makes sense.

We can repurpose `RESET_STATE` as `SET_DEFAULT_STATE` so it makes more sense for either use case (clear and setting an initial empty state)